pkg/client/core/service_application_client.go

```diff
@@ -36,6 +36,7 @@ type applicationService struct {
 	fs                    *afero.Afero
 	absoluteRepositoryDir string
 	kubectl               kubectl.Client
+	gitDeleteRemoteFileFn client.GitDeleteRemoteFileFn
```

Comment:
Consider using an interface instead of passing a function. Using interfaces enables the code reader to simply press CTRL+ALT+B (in intellij) to find the possible implementations. It might not be needed for simple cases, but here we have to backtrack via the New function in service_application_client.go in line 399 before we're able to find out the actual contents of the function - which increases mental load and number of operations in order to find the implementation.